client/src/main/scala/io/delta/sharing/client/DeltaSharingFileSystem.scala

```diff
@@ -69,7 +70,31 @@ private[sharing] class DeltaSharingFileSystem extends FileSystem with Logging {
       val proxy = new HttpHost(proxyConfig.host, proxyConfig.port)
       clientBuilder.setProxy(proxy)
 
-      if (proxyConfig.noProxyHosts.nonEmpty) {
+      val neverUseHttps = ConfUtils.getNeverUseHttps(getConf)
+      if (neverUseHttps) {
```

Comment:
Is this orthogonal with the proxy change? or could it only be applied when a proxy is set?

Reply:
while it's possible to keep the logic separate from the proxy change, it's not realistic (at least from what I can tell) to  downgrade https requests to access storage buckets without one.

Reply:
so it could only be applied when a proxy is set?

Reply:
yes. that is the plan

Reply:
could you guide me in which line is that enforced? 
And what's the code path when there's no proxy set (and it will ignore the `neverUseHttps` config)?

Reply:
if proxy configs (host and port) are not defined, we do not enter [this block](https://github.com/delta-io/delta-sharing/blob/13abc8beb584a947890720aab42e5bd233741d2c/client/src/main/scala/io/delta/sharing/client/DeltaSharingFileSystem.scala#L68). This block contains all of the net-new code introduced in this PR. On top of that, the new code path for attaching the `HttpRequestExecutor` will never get executed if there's a proxy set but `neverUseHttps: false`

Reply:
got it, thanks!